core/extension/gdextension_interface.h

```diff
@@ -420,6 +422,8 @@ typedef struct {
 	GDExtensionCallableCustomLessThan less_than_func;
 
 	GDExtensionCallableCustomToString to_string_func;
+
+	GDExtensionCallableCustomGetArgumentCount get_argument_count_func;
 } GDExtensionCallableCustomInfo;
```

Comment:
I think this is a breaking change:

If someone running a 4.2 extension declares the struct on the binding side, it won't have that last field. If this extension is then run in Godot 4.3, Godot will try to access `get_argument_count_func`, causing UB.

Unfortunately it might need a `GDExtensionCallableCustomInfo2` iteration, but this also impacts other functions 😕 ideas?

Reply:
Then I'll leave the the extension side to future implementation perhaps, wasn't sure hence asking, will look at those details but will split it off to keep it simple on this side 🙂 thank you

Reply:
If there are other similar additions in the future (e.g. advanced argument queries for type/default-ness), it could make sense to batch some changes in a minor release, to avoid proliferation of struct iterations.

Reply:
Indeed, will open a new PR to discuss this more coherently and make a new structure with the related changes there and update the godot-cpp PR, will link here when done

Reply:
I see you added a `GDExtensionScriptInstanceInfo3`, but is this single niche feature really worth another iteration of the struct -- which means extra compat code on both Godot + binding side?

I would really consider holding out until we can pack more into this; otherwise we'll end up with `GDExtensionScriptInstanceInfo7` soon 😉

Reply:
Thinking a bit further I might restrict it to the script side and not on the instance side, we'll see!

Reply:
I have an unrelated change that I've been wanting to make to `GDExtensionScriptInstanceInfo2` - I can try to make a PR for it for 4.3 so this isn't the only change to that struct :-)

Reply:
Then I'll keep this in 🙂

Reply:
Let's try to get @dsnopek's change in with this one for 4.3 then, or keep both (and potential future additions) for 4.4 maybe?

Reply:
Agreed, both at the same time, keeping the extension code in its separate commit until we decide and alternatively splitting it off

Reply:
@akien-mga @AThousandShips Here's a PR with the change I had in mind: https://github.com/godotengine/godot/pull/89055